### ISSUE TITLE
Reusable Build Templates

### ISSUE THREAD

**Issue by grantjk** [czechboy0/Buildasaur#181]
Right now every time I set up a syncer, it asks me to create a new build template for that project.  The majority of the time, however, the PR specs I want to run are the same from project to project. The only thing that is project depending is the scheme. It would be nice if the scheme could be set independent of the template choice so that we can reuse a template across multiple projects.

**Comment by czechboy0** [comment]
Haha, good point. Right now Buildasaur filters the available Build Templates **exactly** because of the scheme, which cannot be reused between projects. 

Pro tip: if you're really creating many of them, go to `~/Library/Application Support/Buildasaur/Build Templates` and duplicate the appropriate JSON files of those Build Templates and just change the `project_name` and `scheme` in them (then quit and relaunch Buildasaur).

**Comment by czechboy0** [comment]
I'm just curious - how many syncers are you creating? :) (The more the happier I'll be haha.)

**Comment by grantjk** [comment]
I have 6 currently - will be probably doubling that over the next few weeks.

**Comment by czechboy0** [comment]
Amazing! But that means you'll eventually start hitting the GitHub rate limit, so #62 will definitely be useful to you. What are your sync intervals?

**Comment by grantjk** [comment]
I just left them at default for now. I'm also super interested in #180 . We currently have 2 servers, but I could see us scaling that up over the next year.

**Comment by grantjk** [comment]
I've also been using "always clean", which means over the past year or so, our first server has downloaded over 355TB of data. :eyes:

**Comment by czechboy0** [comment]
@grantjk Feel free to +1 the issues that you'd also appreciate to be solved sooner rather than later, at the moment I only have a rough idea of which ones are most demanded. (E.g. #180 in your case)

You currently have 2 Xcode Servers, is that what you mean?

And regarding always cleaning, IMHO, from experience, that just _really_ slows down the build and doesn't bring much value (at least decrease it to cleaning once a day).

355TB of data? Wow :laughing:

**Comment by grantjk** [comment]
Good to know! I've followed a bunch of other repos, where people super hate the +1 thing.

Yeah, we have 2 Xcode Servers going. 2 different Mac Minis. Do you know if there's a way to run multiple Xcode servers on a single machine?

Thanks for the tip on the cleaning. I'll try it.

**Comment by czechboy0** [comment]
I don't think you can run multiple Xcode Servers, because it doesn't run under your user. Instead, it's its own user `_xcsbuildd`, so I don't think there's a way to run more than one.

You could do something like having multiple OS X instances running in a VM on a single machine, but otherwise I don't think it's possible.

**Comment by eric-chamberlain-intel** [comment]
We'd like to see this feature too.
